pkg/queue/health/health_state.go

```diff
@@ -92,16 +108,27 @@ func (h *State) HealthHandler(prober func() bool) func(w http.ResponseWriter, r
 	}
 }
 
-// QuitHandler constructs a handler that shuts the current server down.
-// Optional cleanup logic can be added via the given cleaner function.
-func (h *State) QuitHandler(cleaner func()) func(w http.ResponseWriter, r *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		h.shutdown()
+// DrainHandler constructs a handler that waits until the proxy server is shut down.
+func (h *State) DrainHandler() func(_ http.ResponseWriter, _ *http.Request) {
```

Comment:
Considering that this is done once during the construction of the server, why not just create the channel during construction of the `State`? At that point the whole function can be replaced with
```
func (h *State) DrainHandler(_ http.ResponseWriter, _ *http.Request) {
 <- h.drainCh
}
```
and its invocation, as `mux.HandleFunc(queue.RequestQueueDrainPath, healthState.DrainHandler)`